esp-hal/src/soc/esp32s3/mod.rs

```diff
@@ -117,6 +124,19 @@ pub unsafe extern "C" fn ESP32Reset() -> ! {
         addr_of_mut!(_rtc_slow_bss_start),
         addr_of_mut!(_rtc_slow_bss_end),
     );
+    if !matches!(
+        crate::reset::get_reset_reason(),
+        Some(SocResetReason::CoreSw | SocResetReason::CpuSw | SocResetReason::CoreDeepSleep)
+    ) {
+        xtensa_lx_rt::zero_bss(
```

Comment:
Ideally, I thought we should use `init_data` and initialize with valid data but that turns out to be not that easy (would require a bit more linker script magic) - so maybe having it zeroed is the second-best option

Reply:
Looked again into the linker-script and my original idea of actually initializing the segment won't work - so clearing it is the only option